### DISCUSSION TITLE
Configurable max number of pages in pagination feature

### DISCUSSION THREAD

**Discussion by GraphicHealer** [grafana/grafana-infinity-datasource#1050] [Ideas]
Hello there!

I have read #601, and I know the reason you state for the limit "Is to Prevent DDoS and Rate Limiting of API URLs", but that really is not a good reason to limit the capability of this application. Most places will have a DDoS filter in place to begin with, or their Rate Limits are MUCH higher than 5.

Can you please either allow an option to change the rate limit, even if you have to put a BIG FAT WARNING right up on the page stating all the issues it could cause, and even double, or triple check with the user before enabling. You could even hide the setting in an advanced/developer menu.

We aren't asking it to be the default, we're just wanting the option in the first place.
It honestly is one of the biggest things holding this plugin back.

Even if you won't allow it, can you please at least provide a good, reasonable explanation for why you won't enable it?
Referencing #601 and stating "It's to Prevent DDoS and Rate Limiting" just isn't cutting it for anyone anymore.

Thanks!
\- G

(PS: Please DON'T just close this out with a reference to #601, please actually respond. We'd all like some clarification. 😄)

**Comment by yesoreyeram** [comment]
Hi @GraphicHealer - Sorry for the delay getting back. I understand the frustration you are having with the pagination and max pages option and sorry about that.

I wrote a detailed reply in https://github.com/grafana/grafana-infinity-datasource/discussions/601#discussioncomment-11382097 for more visibility. Feel free to provide your thoughts there.

Again. Thanks for trying the infinity plugin and providing feedback.

**Reply by GraphicHealer** [reply]
You are totally fine dude! I had no idea you had given a response. Honestly your explanation makes perfect sense, and I can understand why you would be worried about enabling such a feature. Thank you so much for all your hard work, and thanks for responding to my little comment! 😄
